src/MarginTrading.SettingsService/Controllers/AssetPairsController.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using MarginTrading.SettingsService.Client;
+using MarginTrading.SettingsService.Client.AssetPair;
+using MarginTrading.SettingsService.Client.Enums;
+using MarginTrading.SettingsService.Core.Domain;
+using MarginTrading.SettingsService.Core.Services;
+using MarginTrading.SettingsService.StorageInterfaces.Repositories;
+using Microsoft.AspNetCore.Mvc;
+
+namespace MarginTrading.SettingsService.Controllers
+{
+    /// <summary>
+    /// Asset pairs management
+    /// </summary>
+    [Route("api/assetPairs")]
+    public class AssetPairsController : Controller, IAssetPairsApi
+    {
+        private readonly IAssetPairsRepository _assetPairsRepository;
+        private readonly IConvertService _convertService;
+        private readonly IEventSender _eventSender;
+        
+        public AssetPairsController(
+            IAssetPairsRepository assetPairsRepository,
+            IConvertService convertService, 
+            IEventSender eventSender)
+        {
+            _assetPairsRepository = assetPairsRepository;
+            _convertService = convertService;
+            _eventSender = eventSender;
+        }
+        
+        /// <summary>
+        /// Get the list of asset pairs based on legal entity and matching engine mode
+        /// </summary>
+        /// <param name="legalEntity"></param>
+        /// <param name="matchingEngineMode"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("")]
+        public async Task<List<AssetPairContract>> List([FromQuery] string legalEntity = null, 
+            [FromQuery] MatchingEngineModeContract? matchingEngineMode = null)
+        {
+            Enum.TryParse<MatchingEngineMode>(matchingEngineMode?.ToString(), out var matchingEngineModeDomain);
+            
+            var data = await _assetPairsRepository.GetAsync(assetPair =>
```

Comment:
It is better not to apply filter at all if it does not needed.

Reply:
Condition added